internal/types/scope/scope_test.go

```diff
@@ -51,3 +52,112 @@ func Test_Map(t *testing.T) {
 		})
 	}
 }
+
+func Test_AllowedIn(t *testing.T) {
+	tests := []struct {
+		resource   resource.Type
+		wantScopes []Type
+	}{
+		{
+			resource:   resource.Account,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Alias,
+			wantScopes: []Type{Global},
+		},
+		{
+			resource:   resource.All,
+			wantScopes: []Type{Global, Org, Project},
+		},
+		{
+			resource:   resource.AuthMethod,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.AuthToken,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Billing,
+			wantScopes: []Type{Global},
+		},
+		{
+			resource:   resource.CredentialLibrary,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.Credential,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.CredentialStore,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.Group,
+			wantScopes: []Type{Global, Org, Project},
+		},
+		{
+			resource:   resource.HostCatalog,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.HostSet,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.Host,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.ManagedGroup,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Policy,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Role,
+			wantScopes: []Type{Global, Org, Project},
+		},
+		{
+			resource:   resource.Scope,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.SessionRecording,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Session,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.StorageBucket,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Target,
+			wantScopes: []Type{Project},
+		},
+		{
+			resource:   resource.Unknown,
+			wantScopes: []Type{Unknown},
+		},
+		{
+			resource:   resource.User,
+			wantScopes: []Type{Global, Org},
+		},
+		{
+			resource:   resource.Worker,
+			wantScopes: []Type{Global},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.resource.String(), func(t *testing.T) {
+			assert.Equal(t, tt.wantScopes, AllowedIn(tt.resource))
```

Comment:
Quick fly-by comment: the call here does not match the function signature. Also might be good to add a new test for the `invalid resource type` error path - though you would need to change how you get the test name since the `String` wont work for an unknown type (or just test the invalid resource outside of the test cases

Reply:
Great callouts -- updated the function call here & added a test case for an invalid resource type